crates/nu-cli/tests/completions/mod.rs

```diff
@@ -2289,6 +2289,30 @@ fn extern_custom_completion_short_flag(mut extern_completer: NuCompleter) {
     match_suggestions(&expected, &suggestions);
 }
 
+/// When we're completing the flag name itself, not its value,
+/// custom completions should not be used
+#[rstest]
+fn custom_completion_flag_name_not_value(mut extern_completer: NuCompleter) {
+    let suggestions = extern_completer.complete("spam --f", 8);
+    assert!(
+        suggestions.iter().any(|s| s.value == "--foo"),
+        "Should contain --foo flag"
+    );
+    let should_not_contain: Vec<_> = vec!["cat", "dog", "eel"];
```

Comment:
Do we need to test that cat, dog, eel won't be included in the suggestions? None of them contain `"--f"` anyway